### DISCUSSION TITLE
What is the best way to enable link-time optimization (LTO) on dependencies?

### DISCUSSION THREAD

**Discussion by sweemer** [microsoft/vcpkg#31316] [Q&A] [answered]
This question was previously asked in https://github.com/microsoft/vcpkg/issues/7372 in 2019 and I don't see any updates since then.

As far as I understand, there are at least the following options available:

1. Submit a pull request to enable LTO in all the portfiles that I depend on.
2. Add the necessary compiler flags for LTO to `VCPKG_CXX_FLAGS` in my workspace.
3. Introduce a new variable like `VCPKG_ENABLE_LTO` that I can set in my workspace.

My vote would be for 3 because it seems to be at the right level of abstraction. If others agree, then can this be implemented in vcpkg?

**Comment by Osyotr** [comment, marked as answer]
No, the correct way is 2. (Or provide your own toolchain that enables lto).
For example, see https://github.com/Neumann-A/my-vcpkg-triplets

**Reply by sweemer** [reply]
Thanks for the info!

Eventually I think it would be great to have a [community triplet](https://github.com/microsoft/vcpkg/tree/master/triplets/community) that shows how to enable LTO, assuming other users would find this useful as well.
